Non-unified build fixes, early October 2022 edition

aperezdc opened the pull request:
#### 269b3d2cdb2f4e75088fbd49541ecd2319e9eb7d
<pre>
Non-unified build fixes, early October 2022 edition

Reviewed by Darin Adler.

* Source/WebCore/Modules/mediastream/RTCEncodedAudioFrame.cpp: Add
  missing JavaScriptCore/ArrayBuffer.h inclusion, move main header
  inclusion inside the ENABLE(WEB_RTC) guard to avoid it when WebRTC
  support is disabled.
* Source/WebCore/Modules/mediastream/RTCEncodedVideoFrame.cpp: Ditto.
* Source/WebCore/css/CSSComputedStyleDeclaration.cpp: Add missing
  StyleProperties.h header inclusion.
* Source/WebCore/css/CSSFontFace.h: Add missing WebCore::Document
  forward declaration.
* Source/WebCore/css/StyleSheetContents.cpp: Add missing
  StyleProperties.h header inclusion.
* Source/WebCore/css/parser/CSSParserFastPaths.cpp: Add missing
  CSSProperty.h header inclusion.
* Source/WebCore/css/parser/CSSParserImpl.h: Add missing inclusion of
  the CSSProperty.h header and forward declaration for
  WebCore::StyleRuleContainer.
* Source/WebCore/css/parser/CSSParserObserverWrapper.h: Add missing
  wtf/Vector.h header inclusion.
* Source/WebCore/css/parser/CSSSupportsParser.cpp: Add missing
  StyleRule.h header inclusion.
* Source/WebCore/display/css/DisplayBoxFactory.cpp: Add missing
  CachedImage.h header inclusion.
* Source/WebCore/dom/TreeScope.h: Add missing JSC::JSValue forward
  declaration.
* Source/WebCore/page/Page.cpp: Add missing ScreenOrientationManager.h
  header inclusion.
* Source/WebCore/page/ScreenOrientation.cpp: Add missing Page.h and
  FrameDestructionObserverInlines.h header inclusions.
* Source/WebCore/page/ScreenOrientationType.h: Add missing
  wtf/EnumTraits.h heder inclusion.
* Source/WebCore/rendering/BorderPainter.cpp: Add missing RenderBox.h
  header inclusion.
* Source/WebKit/UIProcess/ProcessThrottler.cpp: Add missing
  wtf/text/TextStream.h header inclusion.
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMHTMLTableSectionElement.cpp:
  Add missing WebCore/HTMLTableRowElement.h header inclusion.
* Source/WebKit/WebProcess/WebCoreSupport/WebScreenOrientationManager.cpp:
  Add missing WebPage.h header inclusion.

Canonical link: <a href="https://commits.webkit.org/255304@main">https://commits.webkit.org/255304@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2895fefa4db43cc1563c3c623179fa330cf87ccc

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/92074 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/1306 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/22589 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/101869 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/161923 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/96074 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/1304 "Built successfully") | [  ~~🛠 mac-debug~~](https://ews-build.webkit.org/#/builders/71/builds/29706 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/84518 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/98045 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/97732 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/819 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/78601 "Built successfully") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/27760 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/82741 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/82338 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/70802 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/36132 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/16352 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/33884 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/17454 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/3673 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/37755 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/40164 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/39651 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/36580 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/dom/TreeScope.h:
```diff
@@ -28,6 +28,7 @@
 
 #include "ExceptionOr.h"
 #include "TreeScopeOrderedMap.h"
+#include <JavaScriptCore/JSCJSValue.h>
```
We try so hard to keep unnecessary header includes out of headers, which over time can help us avoid slowing down build times. Instead of adding this include, I suggest we add a forward declaration:
```
namespace JSC {
class JSValue;
}
```
That should fix compilation without pulling another header into every file that uses `TreeScope.h`.

aperezdc replied:
Of course, typically we prefer to forward-declare types in the headers where possible. In this case `TreeScope.h` returns a plain `JSC::JSValue` so it seemeed like the compiler would need to know the size of the type... but I tried and interestingly enough the forward declaration works in this particular instance 👌🏼

darinadler replied:
That’s my understanding: Compiling a function declaration with a class type as the return type works without the class’s definition. Compiling a call to the function or the function definition requires the class’s definition.

darinadler commented on Source/WebCore/dom/TreeScope.h:
```diff
@@ -33,6 +33,10 @@
 
+namespace JSC {
+class JSValue;
+} // namespace JSC
```
I typically would not add a comment for the end of a tiny namespace section like this.

aperezdc replied:
I'll remove it before landing.

aperezdc committed:
Non-unified build fixes, early October 2022 edition

Reviewed by Darin Adler.

* Source/WebCore/Modules/mediastream/RTCEncodedAudioFrame.cpp: Add
  missing JavaScriptCore/ArrayBuffer.h inclusion, move main header
  inclusion inside the ENABLE(WEB_RTC) guard to avoid it when WebRTC
  support is disabled.
* Source/WebCore/Modules/mediastream/RTCEncodedVideoFrame.cpp: Ditto.
* Source/WebCore/css/CSSComputedStyleDeclaration.cpp: Add missing
  StyleProperties.h header inclusion.
* Source/WebCore/css/CSSFontFace.h: Add missing WebCore::Document
  forward declaration.
* Source/WebCore/css/StyleSheetContents.cpp: Add missing
  StyleProperties.h header inclusion.
* Source/WebCore/css/parser/CSSParserFastPaths.cpp: Add missing
  CSSProperty.h header inclusion.
* Source/WebCore/css/parser/CSSParserImpl.h: Add missing inclusion of
  the CSSProperty.h header and forward declaration for
  WebCore::StyleRuleContainer.
* Source/WebCore/css/parser/CSSParserObserverWrapper.h: Add missing
  wtf/Vector.h header inclusion.
* Source/WebCore/css/parser/CSSSupportsParser.cpp: Add missing
  StyleRule.h header inclusion.
* Source/WebCore/display/css/DisplayBoxFactory.cpp: Add missing
  CachedImage.h header inclusion.
* Source/WebCore/dom/TreeScope.h: Add missing JSC::JSValue forward
  declaration.
* Source/WebCore/page/Page.cpp: Add missing ScreenOrientationManager.h
  header inclusion.
* Source/WebCore/page/ScreenOrientation.cpp: Add missing Page.h and
  FrameDestructionObserverInlines.h header inclusions.
* Source/WebCore/page/ScreenOrientationType.h: Add missing
  wtf/EnumTraits.h heder inclusion.
* Source/WebCore/rendering/BorderPainter.cpp: Add missing RenderBox.h
  header inclusion.
* Source/WebKit/UIProcess/ProcessThrottler.cpp: Add missing
  wtf/text/TextStream.h header inclusion.
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMHTMLTableSectionElement.cpp:
  Add missing WebCore/HTMLTableRowElement.h header inclusion.
* Source/WebKit/WebProcess/WebCoreSupport/WebScreenOrientationManager.cpp:
  Add missing WebPage.h header inclusion.

Canonical link: https://commits.webkit.org/255304@main
